bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVaneerSettingsFactory.java

```diff
@@ -0,0 +1,312 @@
+    }
+    FlowControlSettings.Builder flowControlBuilder = FlowControlSettings.newBuilder();
+    if (maxInflightRpcs > 0) {
+      flowControlBuilder
+          .setMaxOutstandingRequestBytes(bulkOptions.getMaxMemory())
+          .setMaxOutstandingElementCount(maxInflightRpcs * bulkMaxRowKeyCount);
+    }
+
+    batchSettingsBuilder
+        .setIsEnabled(bulkOptions.useBulkApi())
+        .setElementCountThreshold(Long.valueOf(bulkOptions.getBulkMaxRowKeyCount()))
+        .setRequestByteThreshold(bulkOptions.getBulkMaxRequestSize())
+        .setFlowControlSettings(flowControlBuilder.build());
+
+    // TODO(rahulkql): implement bulkMutationThrottling & bulkMutationRpcTargetMs, once available
+    builder.bulkMutationsSettings()
+        .setBatchingSettings(batchSettingsBuilder.build())
+        .setSimpleTimeoutNoRetries(
+            ofMillis(options.getCallOptionsConfig().getShortRpcTimeoutMs()));
+  }
+
+  /**
+   * To build BigtableDataSettings#sampleRowKeysSettings with default retries settings.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param options a {@link BigtableOptions} object.
+   */
+  private static void buildSampleRowKeysSettings(Builder builder, BigtableOptions options) {
+    builder.sampleRowKeysSettings()
+        .setRetrySettings(buildIdempotentRetrySettings(options));
+  }
+
+  /**
+   * To build BigtableDataSettings#mutateRowSettings with default retries settings.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param options a {@link BigtableOptions} object.
+   */
+  private static void buildMutateRowSettings(Builder builder, BigtableOptions options) {
+    builder.mutateRowSettings()
+        .setRetrySettings(buildIdempotentRetrySettings(options));
+  }
+
+  /**
+   * To build BigtableDataSettings#readRowsSettings with default retries settings.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param options a {@link BigtableOptions} object.
+   */
+  private static void buildReadRowsSettings(Builder builder, BigtableOptions options) {
+    RetryOptions retryOptions = options.getRetryOptions();
+
+    RetrySettings.Builder retryBuilder = RetrySettings.newBuilder()
+        .setInitialRetryDelay(ofMillis(retryOptions.getInitialBackoffMillis()))
+        .setRetryDelayMultiplier(retryOptions.getBackoffMultiplier())
+        .setMaxRetryDelay(ofMillis(retryOptions.getMaxElapsedBackoffMillis()))
+        .setMaxAttempts(retryOptions.getMaxScanTimeoutRetries());
+
+    // configurations for RPC timeouts
+    Duration readPartialRowTimeout = ofMillis(retryOptions.getReadPartialRowTimeoutMillis());
+    retryBuilder
+        .setInitialRpcTimeout(readPartialRowTimeout)
+        .setMaxRpcTimeout(readPartialRowTimeout)
+        .setTotalTimeout(ofMillis(options.getCallOptionsConfig().getLongRpcTimeoutMs()));
+
+    builder.readRowsSettings()
+        .setRetrySettings(retryBuilder.build());
+  }
```

Comment:
Please update google-cloud-java dep and add config for single row reads

Reply:
I think we would need readRowSettings() in the [BigtableDataSettings](https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataSettings.java) as well. For now, It is available in [EnhanccedBigtableStubSettings](https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java) only.

Let me know If I should add this change.

Reply:
Please do, I apologize for the omission

Reply:
Added [buildReadRowSettings()](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/1973/files#diff-6a492166d0005302ce3c542cdcf00ad5R174) for point reads.

As of now set it using the same configuration as readRowsSettings.